lib/stack_master/stack_definition.rb

```diff
@@ -81,6 +84,17 @@ def s3_configured?
 
     private
 
+    def s3_template_dir
+      File.dirname(
+        StackMaster.s3_driver.download_file(
+          s3['bucket'],
+          s3['prefix'],
+          s3['region'],
+          template
+          )
+      )
+    end
```

Comment:
It would be worth memoizing this result, to avoid downloading the file everytime we need it.